sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSharedState.scala

```diff
@@ -40,17 +42,22 @@ private[hive] class HiveSharedState(override val sparkContext: SparkContext)
     logInfo(s"Setting Hive metastore warehouse path to '${tempConf.warehousePath}'")
   }
 
+  override lazy val hadoopConf: Configuration = {
+    val conf = sparkContext.hadoopConfiguration
+    conf.addResource("hive-site.xml")
```

Comment:
Yea, we can put it in sql/core. We can use the classloader to find the url and then call add resource.